pkg/scheduler/framework/runtime/framework.go

```diff
@@ -599,24 +599,30 @@ func (f *frameworkImpl) QueueSortFunc() framework.LessFunc {
 
 // RunPreFilterPlugins runs the set of configured PreFilter plugins. It returns
 // *Status and its code is set to non-success if any of the plugins returns
-// anything but Success. If a non-success status is returned, then the scheduling
-// cycle is aborted.
+// anything but Success/Skip.
+// Plugins that returned Skip status are recorded in the cyclestate,
+// and they are skipped in the Filter extension point.
+// If a non-success status is returned, then the scheduling cycle is aborted.
 func (f *frameworkImpl) RunPreFilterPlugins(ctx context.Context, state *framework.CycleState, pod *v1.Pod) (_ *framework.PreFilterResult, status *framework.Status) {
 	startTime := time.Now()
 	defer func() {
 		metrics.FrameworkExtensionPointDuration.WithLabelValues(preFilter, status.Code().String(), f.profileName).Observe(metrics.SinceInSeconds(startTime))
 	}()
 	var result *framework.PreFilterResult
 	var pluginsWithNodes []string
+	skipPlugins := sets.New[string]()
 	for _, pl := range f.preFilterPlugins {
 		r, s := f.runPreFilterPlugin(ctx, pl, state, pod)
-		if !s.IsSuccess() {
+		if !s.IsSuccess() && !s.IsSkip() {
```

Comment:
```golang
if s.IsSkip() {
  skipPlugins.Insert(pl.Name())
} else if !s.IsSuccess() {
  ...
}

Reply:
Actually, wouldn't it be better like this?

```golang
if s.IsSkip() {
  skipPlugins.Insert(pl.Name())
  continue
}
```

A Skip code should not have any relevant information in `r`, right?

Reply:
done

Reply:
Oops, you were too fast. I had another thought above.

Reply:
> A Skip code should not have any relevant information in r, right?

Right. Let me fix

Reply:
Done.